### PR TITLE
[ISSUE-1159]: Disk Eject, disk replacement with fake-attach

### LINKED ISSUE #1159: Disk Eject: drive replacement with fake-attach
**Problem**
The drive-replacement procedure should be performed along with the fake-attach.

**Solution**
Integrated both drive-replacement and fake-attach.

### PR DESCRIPTION
**Pull request by korzepadawid**
### Resolves #1159 

Implemented the disk replacement procedure with the fake attachment.

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [x] New unit tests added
- [x] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Provide test details_

### CODE REVIEW

File: pkg/crcontrollers/drive/drivecontroller.go [hunk 1]
```diff
@@ -311,15 +316,33 @@ func (c *Controller) getVolsStatuses(volumes []*volumecrd.Volume) map[string]str
...
 	}
 	return true
 }
 
+// checkLVGVolumeFakeAttach checks if there is any volume in the provided LVG
+// that is located on LVM and is fake attached, returns true if such a volume is found, false otherwise.
+func (c *Controller) checkLVGVolumeFakeAttach(lvg *lvgcrd.LogicalVolumeGroup, volumes []*volumecrd.Volume) bool {
+	for _, v := range volumes {
+		if v.Spec.LocationType == apiV1.LocationTypeLVM && v.Spec.Location == lvg.Name {
+			return c.isFakeAttach(v)
```
Comment (Andrzej-Zukowski) [inline pkg/crcontrollers/drive/drivecontroller.go:335]: Would it be possible that this function returns false in the following scenario
volumes: [vol1 - fakeattach false, vol2: fakeattach true]

File: pkg/crcontrollers/drive/drivecontroller_test.go [hunk 2]
```diff
@@ -1114,3 +1150,120 @@ func TestDriveController_handleDriveLabelUpdate(t *testing.T) {
...
+						LocationType: apiV1.LocationTypeLVM,
+						Location:     lvgName,
+					},
+					ObjectMeta: objMetaWithFakeAttachAnnotation,
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "Test case where volume is not fake attached",
```
Comment (Andrzej-Zukowski) [inline pkg/crcontrollers/drive/drivecontroller_test.go:1249]: Add test cases with multiple volumes per lvg

File: pkg/node/node.go [hunk 3]
```diff
@@ -259,10 +288,9 @@ func (s *CSINodeService) NodeStageVolume(ctx context.Context, req *csi.NodeStage
 	}
 
 	targetPath := getStagingPath(ll, req.GetStagingTargetPath())
-
-	isFakeAttachNeed := false
+	isFakeAttachNeed, isFakeAttachNeedDR := false, s.isFakeAttachNeededForDR(volumeCR)
```
Comment (Andrzej-Zukowski) [inline pkg/node/node.go:291]: Could you write UT for combined output of mocked isFakeAttachNeededForDR as well as ignoreErrorIfFakeAttach with mocked isPVCNeedFakeAttach. Ideally all possible paths {isFakeAttachNeededForDR, isPVCNeedFakeAttach} = [{false,false}{false,true}{true,false}{true,true}]
Checking whether isFakeAttachNeed and isFakeAttachNeedDR do have expected values

File: pkg/node/node.go [hunk 4]
```diff
@@ -259,12 +298,11 @@ func (s *CSINodeService) NodeStageVolume(ctx context.Context, req *csi.NodeStage
...
-
-	isFakeAttachNeed := false
+	isFakeAttachAnnotation, isFakeAttachDR := s.isFakeAttachNeed(volumeCR)
+	isErr := false
 	ignoreErrorIfFakeAttach := func(err error) {
-		if s.isPVCNeedFakeAttach(volumeID) {
-			isFakeAttachNeed = true
-		} else {
+		isErr = true
+		if !isFakeAttachAnnotation {
```
Comment (korzepadawid) [inline pkg/node/node.go:305]: `isFakeAttachDR` overlaps `isFakeAttachAnnotation`, when `isFakeAttachDR` is `true` `isFakeAttachAnnotation` must be `true` as well

File: pkg/crcontrollers/drive/drivecontroller.go [hunk 5]
```diff
@@ -355,24 +379,54 @@ func (c *Controller) checkAndPlaceStatusRemoved(drive *drivecrd.Drive) bool {
 	return false
 }
 
+func (c *Controller) handleDriveUsageReleasing(ctx context.Context, log *logrus.Entry, drive *drivecrd.Drive) (uint8, error) {
```
Comment (korzepadawid) [inline pkg/crcontrollers/drive/drivecontroller.go:382]: fyi: golint wanted refactor

File: pkg/crcontrollers/drive/drivecontroller.go [hunk 6]
```diff
@@ -165,33 +173,17 @@ func (c *Controller) handleDriveUpdate(ctx context.Context, log *logrus.Entry, d
...
-				break
-			}
-		}
-		if allFound {
-			drive.Spec.Usage = apiV1.DriveUsageReleased
-			eventMsg := fmt.Sprintf("Drive is ready for documented removal procedure. %s", drive.GetDriveDescription())
-			c.eventRecorder.Eventf(drive, eventing.DriveReadyForRemoval, eventMsg)
-			toUpdate = true
-		}
+		return c.handleDriveUsageReleasing(ctx, log, drive)
```
Comment (korzepadawid) [inline pkg/crcontrollers/drive/drivecontroller.go:176]: https://github.com/dell/csi-baremetal/pull/1163/files#r1632613610

**Review by Andrzej-Zukowski — APPROVED** [review]

**Review by rafalw82 — APPROVED** [review]

**Pull request merged**